vc_zoom/indico_vc_zoom/plugin.py

```diff
@@ -0,0 +1,442 @@
+        vc_room.name = zoom_meeting['topic']
+        vc_room.data.update({
+            'description': zoom_meeting['agenda'],
+            'url': zoom_meeting['join_url'],
+            'public_url': zoom_meeting['join_url'].split('?')[0],
+            'zoom_id': zoom_meeting['id'],
+            'password': zoom_meeting['password'],
+            'mute_host_video': zoom_meeting['settings']['host_video'],
+
+            # these options will be empty for webinars
+            'mute_audio': zoom_meeting['settings'].get('mute_upon_entry'),
+            'mute_participant_video': not zoom_meeting['settings'].get('participant_video'),
+            'waiting_room': zoom_meeting['settings'].get('waiting_room')
+        })
+        flag_modified(vc_room, 'data')
+
+    def delete_room(self, vc_room, event):
+        client = ZoomIndicoClient()
+        zoom_id = vc_room.data['zoom_id']
+        is_webinar = vc_room.data['meeting_type'] == 'webinar'
+        try:
+            if is_webinar:
+                client.delete_webinar(zoom_id)
+            else:
+                client.delete_meeting(zoom_id)
+        except HTTPError as e:
+            # if there's a 404, there is no problem, since the room is supposed to be gone anyway
+            if not e.response.status_code == 404:
+                self.logger.exception('Error getting Zoom Room: %s', e.response.content)
+                raise VCRoomError(_("Problem fetching room from Zoom. Please contact support if the error persists."))
+
+    def get_blueprints(self):
+        return blueprint
+
+    def get_vc_room_form_defaults(self, event):
+        defaults = super(ZoomPlugin, self).get_vc_room_form_defaults(event)
+        defaults.update({
+            'meeting_type': 'regular',
+            'mute_audio': self.settings.get('mute_audio'),
+            'mute_host_video': self.settings.get('mute_host_video'),
+            'mute_participant_video': self.settings.get('mute_participant_video'),
+            'waiting_room': self.settings.get('waiting_room'),
+            'host_choice': 'myself',
+            'host_user': None,
+            'password_visibility': 'logged_in'
+        })
+        return defaults
+
+    def get_vc_room_attach_form_defaults(self, event):
+        defaults = super(ZoomPlugin, self).get_vc_room_attach_form_defaults(event)
+        defaults['password_visibility'] = 'logged_in'
+        return defaults
+
+    def can_manage_vc_room(self, user, room):
+        return (
+            user == principal_from_identifier(room.data['host']) or
+            super(ZoomPlugin, self).can_manage_vc_room(user, room)
+        )
+
+    def _merge_users(self, target, source, **kwargs):
+        super(ZoomPlugin, self)._merge_users(target, source, **kwargs)
+        for room in VCRoom.query.filter(
+            VCRoom.type == self.service_name, VCRoom.data.contains({'host': source.identifier})
+        ):
+            room.data['host'] = target.id
```

Comment:
Are you sure this is correct? You filter by `source.identifier` but set it to `target.id`

Reply:
Indeed. Bug. Well spotted.